src/api/app/models/kiwi/image.rb

```diff
@@ -143,6 +143,11 @@ def default_package_group
     package_groups.find_or_create_by(kiwi_type: :image, pattern_type: 'onlyRequired')
   end
 
+  def self.map_to_allowed_repository_types(repository)
```

Comment:
I think we should move this to the repository model as it the method name already suggests that this is the wrong place, no?